:memo: update guide documentation (#157)
mindee/mindee-api-ruby: docs/common_file_operations.md

### Polling options

When performing an asynchronous parse (i.e. when the document is enqueued), the client will poll the API for the result. The following options control the polling behavior:

- `initial_delay_sec`: The initial delay (in seconds) before the first polling attempt.
- `delay_sec`: The delay (in seconds) between subsequent polls.
- `max_retries`: The maximum number of polling attempts before timing out.

These parameters ensure that the client does not overload the API with too-frequent requests and also avoid premature timeouts.

Example:
```ruby
result = mindee_client.parse(
  input_source,
  Mindee::Product::Invoice::InvoiceV4,
  options: {
    initial_delay_sec: 2,   # Wait 2 seconds before the first poll.
    delay_sec: 1.5,         # Wait 1.5 seconds between polls.
    max_retries: 80         # Try polling a maximum of 80 times.
  }
)
```
> ⚠️ Warning: Setting `delay_sec` too low might lead to insufficient wait time between polls, causing the server to block your API calls for a short time.

> 

Reviewer: not sure this belongs in "file operations", would be better under "getting started" IMHO

Author: It was starting to get unreadable due to the amount of information in it, I'd like to take as much stuff out as possible

Author: Will create a new page :+1:

> Because the process involves re-rendering the PDF’s contents, some source text may be lost or rendered differently.
> Use this feature with caution.

```ruby
# Load a local input source.
input_file_path = "path/to/your/file.pdf"
output_file_path = "path/to/the/compressed/file.pdf"
pdf_input = Mindee::Input::Source::PathInputSource.new(input_file_path)

# We advise you test the quality value yourself, as results may vary greatly depending on the input file
pdf_input.compress!(quality: 50)

# Write the output file locally for visual checking:
File.write(output_file_path, pdf_input.io_stream.read)
```

> 🚧 Be warned that the source text (the text embedded in the PDF itself) might not render properly,
> and so source PDFs will be ignored by default.
> You can bypass this using:

```ruby
pdf_input.compress!(quality: 50, force_source_text: true)
```

Or alternatively, you can try to approximate the re-rendering of the source-text using:

```ruby
pdf_input.compress!(quality: 50, force_source_text: true, disable_source_text: false)
```

> #### PDF fixing

Reviewer: ```suggestion
#### PDF Repair
```

> The PDF fixing feature helps to rescue PDFs with invalid or broken header information.

Reviewer: ```suggestion
The PDF repair feature attempts to rescue PDFs with invalid or broken header information.
```

This can sometimes help when files get rejected by the server.

Example:
```ruby
# Load a PDF file with the fix_pdf flag enabled.

> input_source = mindee_client.source_from_file(file, "document.pdf", fix_pdf: true)

Reviewer: possibly change the wording of the feature to `repair_pdf: true`

```

> > ⚠️ Warning: PDF fixing alters the input file by re-writing header information.

Reviewer: ```suggestion
> ⚠️ Warning: PDF repair alters the input file by re-writing header information.
```
